chain/vm/vm.go

```diff
@@ -596,17 +599,27 @@ func copyRec(from, to blockstore.Blockstore, root cid.Cid, cp func(block.Block)
 			return
 		}
 
-		if link.Prefix().MhType == mh.IDENTITY || link.Prefix().Codec == cid.FilCommitmentSealed || link.Prefix().Codec == cid.FilCommitmentUnsealed {
+		prefix := link.Prefix()
+		if prefix.Codec == cid.FilCommitmentSealed || prefix.Codec == cid.FilCommitmentUnsealed {
 			return
 		}
 
-		has, err := to.Has(link)
-		if err != nil {
-			lerr = xerrors.Errorf("has: %w", err)
-			return
-		}
-		if has {
-			return
+		// We always have blocks inlined into CIDs, but we may not have their children.
+		if prefix.MhType == mh.IDENTITY {
+			// Unless the inlined block has no children.
+			if prefix.Codec == cid.Raw {
+				return
+			}
+		} else {
+			// If we have an object, we already have its children, skip the object.
+			has, err := to.Has(link)
+			if err != nil {
+				lerr = xerrors.Errorf("has: %w", err)
+				return
+			}
+			if has {
+				return
+			}
```

Comment:
This is a semi-related bug fix just in case we happen to find a link in a block inlined into a CID. Does this code make sense?

Reply:
Y.. Yes

This probably wasn't an issue before because we didn't do automatic inlining